LuaUI/Widgets/gui_defenseRange.lua

```diff
@@ -448,33 +479,66 @@ function widget:Update(dt)
 	end
 end
 
+local function DoFullUnitReload()
+	for unitID,def in pairs(defences) do
+		RemoveUnit(unitID)
+	end
+	local myAllyTeam = Spring.GetMyAllyTeamID()
+	local units = Spring.GetAllUnits()
+	for i = 1, #units do
+		local unitID = units[i]
+		local unitAllyTeam = Spring.GetUnitAllyTeam(unitID)
+		UnitDetected(unitID, Spring.GetUnitDefID(unitID), unitAllyTeam == myAllyTeam, true)
+	end
+end
+
 function widget:GameFrame(n)
 	if (n % 30) ~= 0 then
 		return
 	end
 
-	for unitID, def in pairs(defences) do
+	for unitID in pairs(defNeedingBuildingChecks) do
 		local unitDefID = spGetUnitDefID(unitID)
 		if unitDefID then
-			if defences[unitID].checkCompleteness then
-				-- Not allied because this check is only required for enemies.
-				-- Allied units are detected in UnitFinished.
-				UnitDetected(unitID, unitDefID, false)
+			UnitDetected(unitID, unitDefID, false)
+		end
+	end
+
+	for unitID in pairs(defNeedingLosChecks) do
+		if not spGetUnitDefID(unitID) then
+			local def = defences[unitID]
+			if select(2, spGetPositionLosState(def.x, def.y, def.z)) then
```

Comment:
select is slightly slower than locals 
```suggestion
			local _, inLos = spGetPositionLosState(def.x, def.y, def.z)
			if inLos then
```